pubsub/pulsar/pulsar.go

```diff
@@ -123,16 +123,30 @@ func (p *Pulsar) Publish(req *pubsub.PublishRequest) error {
 		producer = cache.(pulsar.Producer)
 	}
 
-	_, err = producer.Send(context.Background(), &pulsar.ProducerMessage{
-		Payload: req.Data,
-	})
-	if err != nil {
+	msg := parsePublishMetadata(req)
+	if _, err = producer.Send(context.Background(), msg); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// parsePublishMetadata parse publish metadata
+func parsePublishMetadata(req *pubsub.PublishRequest) (msg *pulsar.ProducerMessage) {
+	msg = &pulsar.ProducerMessage{
+		Payload: req.Data,
+	}
+	if val, ok := req.Metadata["deliver_at"]; ok {
```

Comment:
I would follow the metadata naming convention in Dapr for `camelCased` metadata property names, so `deliverAt` and `deliverAfter`.